src/tiled/layeroffsettool.cpp

```diff
@@ -153,7 +153,8 @@ void LayerOffsetTool::finishDrag()
     if (!mapDocument())
         return;
 
-    if (Layer *layer = mapDocument()->currentLayer()) {
+    Layer *layer = mapDocument()->currentLayer();
+    if (layer && layer->isUnlocked()) {
```

Comment:
This check is too late (as is the check in `mouseMoved`). Rather, `startDrag` should refuse to start dragging when the layer is locked.